scripts/setup.py

```
@@ -34,95 +34,104 @@ def init_nx(nx_python_plugin_version: str):
+        print(f"❌ Error: Failed to create virtual environment with Python {python_version}.")
+        print(f"💡 This usually means Python {python_version} is not installed or not found by uv.")
+        print(f"💡 Please install Python {python_version} and try again:")
+        print(f"   - Using uv: uv python install {python_version}")
+        print(f"   - Using pyenv: pyenv install {python_version}")
+        print(f"   - Or install from https://python.org")
+        sys.exit(1)
+
+    # Sync dependencies based on profile
+    if profile == "core":
+        print("Syncing core dependencies only...")
+        sync_dependencies(profile)
```

Ruff (0.12.2)

Undefined name `sync_dependencies`

(F821)